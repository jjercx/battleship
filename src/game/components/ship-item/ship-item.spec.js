import React from "react";
import ShipItem from "./ship-item";
import renderer from "react-test-renderer";
import Ship from "game/models/ship";

describe("<ShipItem />", () => {
  describe("without ship", () => {
    it("matches snapshot", () => {
      const tree = renderer.create(<ShipItem />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("with ship", () => {
    it("matches snapshot", () => {
      const ship = new Ship({ size: 4 });
      const tree = renderer.create(<ShipItem ship={ship} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
