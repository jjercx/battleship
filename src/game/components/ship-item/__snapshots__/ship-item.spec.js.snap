// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShipItem /> with ship matches snapshot 1`] = `
<div
  className="sc-AxirZ jlJmpf"
>
  <img
    className="sc-AxjAm decyKT ship-image"
    src="aircraft.png"
  />
  <div
    className="sc-AxiKw VdQH"
  />
  <div
    className="sc-AxiKw VdQH"
  />
  <div
    className="sc-AxiKw VdQH"
  />
  <div
    className="sc-AxiKw VdQH"
  />
</div>
`;

exports[`<ShipItem /> without ship matches snapshot 1`] = `null`;
